subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -84,12 +84,11 @@ public void execute(TaskInternal task, TaskStateInternal state, TaskExecutionCon
     private GradleException executeActions(TaskInternal task, TaskStateInternal state, TaskExecutionContext context) {
         LOGGER.debug("Executing actions for {}.", task);
         final List<ContextAwareTaskAction> actions = new ArrayList<ContextAwareTaskAction>(task.getTaskActions());
-        int actionNumber = 1;
         for (ContextAwareTaskAction action : actions) {
             state.setDidWork(true);
             task.getStandardOutputCapture().start();
             try {
-                executeAction("Execute task action " + actionNumber + "/" + actions.size() + " for " + task.getPath(), task, action, context);
+                executeAction(action.getDisplayName(), task, action, context);
```

Comment:
I did it this way such that specific actions have full control over their display name. At the moment for example `Validate task inputs` (instead of `Execute validate task inputs`). But given that these specific actions are all candidates for removal, it is probably also fine to prepend `Execute ` to every action.